src/main/java/seedu/address/MainApp.java

```diff
@@ -90,7 +95,18 @@ private Model initModelManager(Storage storage, ReadOnlyUserPrefs userPrefs) {
             initialData = new EduMate();
         }
 
-        return new ModelManager(initialData, userPrefs);
+        try {
+            eduMateHistoryOptional = storage.readEduMateHistory();
+            if (!eduMateHistoryOptional.isPresent()) {
+                logger.info("Data file not found. Will be starting with an empty EduMateHistory");
+            }
+            eduMateHistory = eduMateHistoryOptional.orElseGet(EduMateHistory::new);
+        } catch (IOException e) {
+            logger.warning("Problem while reading from the file. Will be starting with an empty EduMateHistory");
+            eduMateHistory = new EduMateHistory();
+        }
```

Comment:
Glad you did not use exception handling as control flow as taught in CS2030S! Keep it up!